File: api/app/schemas/drivers.py
from __future__ import annotations

from uuid import UUID

from pydantic import BaseModel, Field


class DriverCreateItem(BaseModel):
    display_name: str = Field(min_length=1)
    team_id: UUID | None = None


class DriverBulkCreate(BaseModel):
    items: list[DriverCreateItem] = Field(min_length=1)


class DriverUpdate(BaseModel):
    display_name: str | None = Field(default=None, min_length=1)
    team_id: UUID | None = Field(default=None)


class DriverRead(BaseModel):
    id: UUID
    league_id: UUID
    display_name: str
    user_id: UUID | None
    discord_id: str | None
    team_id: UUID | None
    team_name: str | None

    class Config:
        from_attributes = True

File: frontend/src/types/drivers.ts
export interface DriverRead {
  id: string
  display_name: string
  team_id?: string | null
  team_name?: string | null
  user?: {
    id: string
    discord_username: string | null
  } | null
  discord_id?: string | null
}

export interface DriverSummary {
  id: string
  displayName: string
  teamId: string | null
  teamName: string | null
  linkedUser: boolean
  discordId: string | null
  userName: string | null
}

export interface BulkDriverInput {
  display_name: string
  team_id?: string | null
}

export interface UpdateDriverRequest {
  display_name?: string
  team_id?: string | null
}
